### ISSUE TITLE
Makefile does not spot changes in cpp-common

### ISSUE THREAD

**Issue by AlexHockey** [Metaswitch/clearwater-snmp-handlers#41]
Example of building `alarm_handler.so`

```
alex@pc4235-mint ~/src/clearwater-snmp-handlers (master) $ make alarm_handler.so 
g++ `net-snmp-config --cflags` -Wall -std=c++0x -g -O0 -fPIC -shared -I/home/alex/src/clearwater-snmp-handlers/modules/cpp-common/include -o alarm_handler.so alarm_handler.cpp modules/cpp-common/src/alarmdefinition.cpp alarm_table_defs.cpp alarm_model_table.cpp alarm_req_listener.cpp alarm_trap_sender.cpp itu_alarm_table.cpp `net-snmp-config --libs` -lzmq -lpthread
alex@pc4235-mint ~/src/clearwater-snmp-handlers (master) $ touch modules/cpp-common/src/alarmdefinition.cpp 
alex@pc4235-mint ~/src/clearwater-snmp-handlers (master) $ make alarm_handler.so 
/home/alex/src/clearwater-snmp-handlers/build/obj/handler_test/alarmdefinition.depends
/home/alex/src/clearwater-snmp-handlers/build/obj/handler/alarmdefinition.depends
make: `alarm_handler.so' is up to date.
alex@pc4235-mint ~/src/clearwater-snmp-handlers (master) $ 
```

Here's the `alarm_handler.so` line from the Makefile. notice it does depend on `alarmdefinition.cpp`

```
alarm_handler.so: *.cpp *.hpp
	g++ `net-snmp-config --cflags` -Wall -std=c++0x -g -O0 -fPIC -shared ${ALARM_INCLUDES} -o alarm_handler.so alarm_handler.cpp modules/cpp-common/src/alarmdefinition.cpp alarm_table_defs.cpp alarm_model_table.cpp alarm_req_listener.cpp alarm_trap_sender.cpp itu_alarm_table.cpp `net-snmp-config --libs` -lzmq -lpthread
```

**Comment by AlexHockey** [comment]
This could cause us to deliver out-of-date binaries. It's not a massive problem for PC since Jenkins does a clean rebuild each time, but this still feels like a medium priority bug.

**Comment by AlexHockey** [comment]
Whoever takes this should try and fix up #21 at the same time.
